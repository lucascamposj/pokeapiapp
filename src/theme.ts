// theme.ts
import { DefaultTheme } from 'styled-components/native';

export const defaultTheme: DefaultTheme = {
  colors: {
    common: {
      red: '#f00',
      white: '#ffffff',
      gray: '#333',
      black: '#000',
    },
    primary: {
      main: '#f00',
      contrastText: '#fff',
    },
    secondary: {
      main: '#fff',
      contrastText: '#000',
    },
  },
};
