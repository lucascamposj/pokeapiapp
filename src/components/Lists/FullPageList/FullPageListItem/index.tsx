import React from 'react';
import { GestureResponderEvent, TouchableOpacityProps } from 'react-native';

import { Container, ItemText } from './styles';

export interface ListItemProps extends TouchableOpacityProps {
  name: string;
  url: string;
  onPressItem: (
    event: GestureResponderEvent,
    name: string,
    url: string,
  ) => void;
}

const FullPageListItem: React.FC<ListItemProps> = ({
  name,
  url,
  onPressItem,
  ...rest
}: ListItemProps) => {
  return (
    <Container onPress={(e) => onPressItem(e, name, url)} {...rest}>
      <ItemText>{name}</ItemText>
    </Container>
  );
};

export default FullPageListItem;
