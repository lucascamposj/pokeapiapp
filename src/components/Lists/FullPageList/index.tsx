import React from 'react';
import { ActivityIndicator, GestureResponderEvent } from 'react-native';
import {} from 'styled-components/native';
import FullPageListItem from './FullPageListItem';
import {
  FlatListStyled as FlatList,
  FooterContainer,
  FooterText,
} from './styles';

interface ListProps {
  list: any[];
  loading: boolean;
  onEndReached: () => void | undefined;
  onPressItem: (
    event: GestureResponderEvent,
    name: string,
    url: string,
  ) => void;
}

const FullPageList: React.FC<ListProps> = ({
  list,
  onEndReached,
  loading = false,
  onPressItem,
}: ListProps) => {
  return (
    <FlatList
      data={list}
      keyExtractor={({ name }) => name}
      renderItem={({ item: { name, url } }) => {
        return (
          <FullPageListItem name={name} url={url} onPressItem={onPressItem} />
        );
      }}
      ListFooterComponent={
        <FooterContainer>
          {loading ? (
            <ActivityIndicator
              color="red"
              size="large"
              style={{ flex: 1, marginTop: 20 }}
            />
          ) : (
            <FooterText>Gotta catch them all!</FooterText>
          )}
        </FooterContainer>
      }
      onEndReached={onEndReached}
      onEndReachedThreshold={0.9}
    />
  );
};

export default FullPageList;
