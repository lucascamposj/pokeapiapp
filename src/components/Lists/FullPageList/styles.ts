import styled from 'styled-components/native';
import { FlatList } from 'react-native';
import { ListItemProps } from './FullPageListItem';

export const FlatListStyled = styled(
  FlatList as new () => FlatList<ListItemProps>,
)`
  padding: 16px 24px 16px;
`;

export const FooterContainer = styled.View`
  flex: 1;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: 70px;
`;

export const FooterText = styled.Text`
  text-align: center;
  font-size: 20px;
  margin: 10px 0;
`;
