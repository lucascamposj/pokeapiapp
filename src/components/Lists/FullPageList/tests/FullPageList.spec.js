import React from 'react';
import {shallow, mount} from 'enzyme';
import {ThemeProvider} from 'styled-components/native';
import {defaultTheme} from '../../../../theme';
import FullPageList from '../../FullPageList';
import 'jest-styled-components';

// executes shallow with ThemeProvider as context
export function shallowWithTheme(child) {
  return shallow(child, {
      wrappingComponent: ({ children }) => <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>,
  });
}

describe('FullPageListItem', () => {
    describe('Rendering', () => {
        it('should match to snapshot loaded', () => {
            const list = [
              {
                name: "Pikachu",
                url: "pikachu.com.br"
              },
              {
                name: "Bulbasaur",
                url: "bulbasaur.com.br"
              }
            ]
            const mockEndReachedHandler = jest.fn();
            const mockPressItem = jest.fn();

            const component = shallowWithTheme(
              <FullPageList
                list={list}
                loading={false}
                onEndReached={mockEndReachedHandler}
                onPressItem={mockPressItem}
              />)
            expect(component).toMatchSnapshot()
        });

        it('should match to snapshot loading', () => {
          const list = [
            {
              name: "Pikachu",
              url: "pikachu.com.br"
            },
            {
              name: "Bulbasaur",
              url: "bulbasaur.com.br"
            }
          ]
          const mockEndReachedHandler = jest.fn();
          const mockPressItem = jest.fn();

          const component = shallowWithTheme(
            <FullPageList
              list={list}
              loading={true}
              onEndReached={mockEndReachedHandler}
              onPressItem={mockPressItem}
            />)
          expect(component).toMatchSnapshot()
        });

        it('should match to snapshot with loading prop undefined ', () => {
          const list = [
            {
              name: "Pikachu",
              url: "pikachu.com.br"
            },
            {
              name: "Bulbasaur",
              url: "bulbasaur.com.br"
            }
          ]
          const mockEndReachedHandler = jest.fn();
          const mockPressItem = jest.fn();

          const component = shallowWithTheme(
            <FullPageList
              list={list}
              onEndReached={mockEndReachedHandler}
              onPressItem={mockPressItem}
            />)
          expect(component).toMatchSnapshot()
        });
    });
});
