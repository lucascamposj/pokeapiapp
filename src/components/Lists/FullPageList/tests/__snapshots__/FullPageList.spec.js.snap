// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullPageListItem Rendering should match to snapshot loaded 1`] = `
<Styled(FlatList)
  ListFooterComponent={
    <Styled(View)>
      <Styled(Text)>
        Gotta catch them all!
      </Styled(Text)>
    </Styled(View)>
  }
  data={
    Array [
      Object {
        "name": "Pikachu",
        "url": "pikachu.com.br",
      },
      Object {
        "name": "Bulbasaur",
        "url": "bulbasaur.com.br",
      },
    ]
  }
  keyExtractor={[Function]}
  onEndReached={[MockFunction]}
  onEndReachedThreshold={0.9}
  renderItem={[Function]}
/>
`;

exports[`FullPageListItem Rendering should match to snapshot loading 1`] = `
<Styled(FlatList)
  ListFooterComponent={
    <Styled(View)>
      <ActivityIndicator
        animating={true}
        color="red"
        hidesWhenStopped={true}
        size="large"
        style={
          Object {
            "flex": 1,
            "marginTop": 20,
          }
        }
      />
    </Styled(View)>
  }
  data={
    Array [
      Object {
        "name": "Pikachu",
        "url": "pikachu.com.br",
      },
      Object {
        "name": "Bulbasaur",
        "url": "bulbasaur.com.br",
      },
    ]
  }
  keyExtractor={[Function]}
  onEndReached={[MockFunction]}
  onEndReachedThreshold={0.9}
  renderItem={[Function]}
/>
`;

exports[`FullPageListItem Rendering should match to snapshot with loading prop undefined  1`] = `
<Styled(FlatList)
  ListFooterComponent={
    <Styled(View)>
      <Styled(Text)>
        Gotta catch them all!
      </Styled(Text)>
    </Styled(View)>
  }
  data={
    Array [
      Object {
        "name": "Pikachu",
        "url": "pikachu.com.br",
      },
      Object {
        "name": "Bulbasaur",
        "url": "bulbasaur.com.br",
      },
    ]
  }
  keyExtractor={[Function]}
  onEndReached={[MockFunction]}
  onEndReachedThreshold={0.9}
  renderItem={[Function]}
/>
`;
